File: api/app/schemas/connector.py
from datetime import datetime
from typing import Any, Dict, Optional

from pydantic import BaseModel, ConfigDict


class PairingCodeResponse(BaseModel):
    connector_id: str
    pairing_code: str
    status: str


class ConnectorStatusResponse(BaseModel):
    connector_id: Optional[str] = None
    status: str
    last_seen_at: Optional[datetime] = None


class ConnectorJobRequest(BaseModel):
    connection_string: str


class ConnectorQueryRequest(ConnectorJobRequest):
    sql: str


class ConnectorConnectionTestResponse(BaseModel):
    status: str
    message: str


class ConnectorSchemaResponse(BaseModel):
    status: str
    tables_count: int
    schema: Dict[str, Any]


class ConnectorQueryResponse(BaseModel):
    status: str
    rows: list[Dict[str, Any]]
    result_profile: Dict[str, Any]


class ConnectorResponse(BaseModel):
    id: str
    name: str
    status: str
    last_seen_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/connector.ts
import { ConnectionTestResponse, DatabaseSchemaResponse } from './sql';

export interface PairingCodeResponse {
  connector_id: string;
  pairing_code: string;
  status: string;
}

export interface ConnectorStatusResponse {
  connector_id?: string | null;
  status: 'pending' | 'online' | 'offline';
  last_seen_at?: string | null;
}

export type ConnectorConnectionTestResponse = ConnectionTestResponse;
export type ConnectorSchemaResponse = DatabaseSchemaResponse;
